app/src/main/java/org/mozilla/focus/experiments/NimbusSetup.kt

```diff
@@ -34,87 +38,76 @@ fun createNimbus(context: Context, url: String?): NimbusApi {
+        channel = BuildConfig.BUILD_TYPE,
+    )
+    return try {
         Nimbus(context, appInfo, serverSettings, errorReporter).apply {
-            val isTheFirstLaunch = context.settings.getAppLaunchCount() == 0
-            if (isTheFirstLaunch) {
-                register(EventsObserver)
-            }
+            register(EventsObserver)
 
-            // This performs the minimal amount of work required to load branch and enrolment data
-            // into memory. If `getExperimentBranch` is called from another thread between here
-            // and the next nimbus disk write (setting `globalUserParticipation` or
-            // `applyPendingExperiments()`) then this has you covered.
-            // This call does its work on the db thread.
-            initialize()
-
-            if (!enabled) {
-                // This opts out of nimbus experiments. It involves writing to disk, so does its
-                // work on the db thread.
-                globalUserParticipation = enabled
-            }
-
-            if (url.isNullOrBlank()) {
-                setExperimentsLocally(R.raw.initial_experiments)
+            val job = if (isTheFirstLaunch || url.isNullOrBlank()) {
+                applyLocalExperiments(R.raw.initial_experiments)
+            } else {
+                applyPendingExperiments()
             }
 
-            if (isTheFirstLaunch) {
-                NimbusExperiments.nimbusInitialFetch.start()
+            runBlocking {
+                job.joinOrTimeout(TIME_OUT_LOADING_EXPERIMENT_FROM_DISK_MS)
             }
-            fetchExperiments()
+        }
+    } catch (e: Throwable) {
+        // Something went wrong. We'd like not to, but stability of the app is more important than
+        // failing fast here.
+        errorReporter("Failed to initialize Nimbus", e)
+        NimbusDisabled(context = context)
+    }
+}
 
+internal fun finishNimbusInitialization(experiments: NimbusApi) =
+    experiments.run {
+        // We fetch experiments in all cases,
+        if (context.settings.getAppLaunchCount() == 0) {
+            // … however on first run, we immediately apply pending experiments.
+            // We also want to measure how long this will take, with Glean.
             register(
                 object : NimbusInterface.Observer {
                     override fun onExperimentsFetched() {
+                        NimbusExperiments.nimbusInitialFetch.stop()
                         applyPendingExperiments()
-                        if (isTheFirstLaunch) {
-                            NimbusExperiments.nimbusInitialFetch.stop()
-                        }
                         // Remove lingering observer when we're done fetching experiments on startup.
                         unregister(this)
                     }
                 },
             )
+            NimbusExperiments.nimbusInitialFetch.start()
         }
```

Comment:
If we want to switch the bundling the initial_experiments, we'd want to remove these lines, add the experiment download to the repo Github Actions workflow, and disable the splash screen.